kernel/arch/dreamcast/hardware/biosfont.c

```diff
@@ -285,29 +290,31 @@ unsigned char bfont_draw_ex(uint8 *buffer, uint32 bufwidth, uint32 fg, uint32 bg
 }
 
 /* Draw half-width kana */
-unsigned char bfont_draw_thin(void *b, uint32 bufwidth, uint8 opaque, uint32 c, uint8 iskana) {
-    return bfont_draw_ex((uint8 *)b, bufwidth, bfont_fgcolor, bfont_bgcolor, (bfont_32bit ? (sizeof (uint32)) : (sizeof (uint16))) << 3, opaque, c, 0, iskana);
+size_t bfont_draw_thin(void *b, uint32_t bufwidth, bool opaque, uint32_t c, bool iskana) {
+    return bfont_draw_ex((uint8_t *)b, bufwidth, bfont_fgcolor, bfont_bgcolor, 
+                         (bfont_32bit ? (sizeof (uint32_t)) : (sizeof (uint16_t))) << 3, opaque, c, 0, iskana);
 }
 
 /* Compat function */
-unsigned char bfont_draw(void *buffer, uint32 bufwidth, uint8 opaque, uint32 c) {
-    return bfont_draw_ex((uint8 *)buffer, bufwidth, bfont_fgcolor, bfont_bgcolor, (bfont_32bit ? (sizeof (uint32)) : (sizeof (uint16))) << 3, opaque, c, 0, 0);
+size_t bfont_draw(void *buffer, uint32_t bufwidth, bool opaque, uint32_t c) {
+    return bfont_draw_ex((uint8_t *)buffer, bufwidth, bfont_fgcolor, bfont_bgcolor, 
+                        (bfont_32bit ? (sizeof (uint32_t)) : (sizeof (uint16_t))) << 3, opaque, c, 0, 0);
 }
 
 /* Draw wide character */
-unsigned char bfont_draw_wide(void *b, uint32 bufwidth, uint8 opaque, uint32 c) {
-    return bfont_draw_ex((uint8 *)b, bufwidth, bfont_fgcolor, bfont_bgcolor, (bfont_32bit ? (sizeof (uint32)) : (sizeof (uint16))) << 3, opaque, c, 1, 0);
+size_t bfont_draw_wide(void *b, uint32_t bufwidth, bool opaque, uint32_t c) {
+    return bfont_draw_ex((uint8_t *)b, bufwidth, bfont_fgcolor, bfont_bgcolor, 
+                         (bfont_32bit ? (sizeof (uint32_t)) : (sizeof (uint16_t))) << 3, opaque, c, 1, 0);
 }
 
-/* Draw string of full-width (wide) and half-width (thin) characters
-   Note that this handles the case of mixed encodings unless Japanese
-   support is disabled (BFONT_CODE_ISO8859_1).
-   XXX: Seems like this can be shrunk to use uint8 for nChr/Mask/Flag and
-    getting rid of nMask.
-   */
-void bfont_draw_str_ex(void *b, uint32 width, uint32 fg, uint32 bg, uint8 bpp, uint8 opaque, const char *str) {
-    uint16 nChr, nMask, nFlag;
-    uint8 *buffer = (uint8 *)b;
+void bfont_draw_str_ex_va(void *b, uint32_t width, uint32_t fg, uint32_t bg, uint8_t bpp, bool opaque, 
+                          const char *fmt, va_list *var_args) {
+    char string[1024 * 4];
```

Comment:
Just wondering if there's any specific rationale for the size of `string` here.

Reply:
I think the screen can fit a total of 1060 of the thin width bios font characters.

Reply:
Ah, good call. I wasn't sure how many could fit onscreen, so that seemed okay, but what @andressbarajas is suggesting is far more precise.

Reply:
> Ah, good call. I wasn't sure how many could fit onscreen, so that seemed okay, but what @andressbarajas is suggesting is far more precise.

So I wasn't sure and dug in more. C99 standard says the max size for a printf buffer should be at least 4095, so maybe that was in your mind when you picked it. :) I wonder if there's a define to pull that kind of thing out of newlib.